KMB/kmb_massload.py

```diff
@@ -169,34 +169,46 @@ def process_byline(entry):
         entry['byline'] = helpers.flip_name(entry['byline'])
 
 
-# @todo: update this per new copyright rules - T164568
 def process_license(entry):
     """
-    Identify the license and store back in entry.
+    Identify the license and store the back in entry.
 
     Don't include name/byline if unknown.
     """
+    entry['copyright'] = entry['copyright'].strip()
+    license_text = None
+
     if entry['license']:
         trim = 'http://kulturarvsdata.se/resurser/License#'
         entry['license'] = entry['license'].strip()[len(trim):]
-    # if not copyright = RAÄ and if license='' then, this is probably unfree
+
     if (entry['license'] == 'pdmark') or \
-            (entry['copyright'].strip() == 'Utgången upphovsrätt'):
-        entry['license'] = '{{PD-Sweden-photo}}'
-    elif (entry['license'] == 'by') or (entry['copyright'].strip() == 'RAÄ'):
-        # consider changing this to AND since there might be a cc-by image which isn't from RAA.
-        # Alternatively have another if inside which checks whethere copyright = RAA
-        param = '}}'
-        if (entry['byline'] == '{{unknown}}') or \
-                (entry['byline'] == '{{not provided}}'):
-            pass
-        else:
-            param = '|%s}}' % entry['byline']
-        entry['license'] = '{{CC-BY-RAÄ%s' % param
+            (entry['copyright'] == 'Utgången upphovsrätt'):
+        license_text = '{{PD-Sweden-photo}}'
+    elif entry['license'] == 'by':
+        byline = None
+        template = 'CC-BY-SA-2.5'
+        if entry['byline'] not in ('{{unknown}}', '{{not provided}}'):
+            byline = '|{0}'.format(entry['byline'])
+
+        if entry['copyright'] == 'RAÄ':
+            template = 'CC-BY-RAÄ'
+        elif entry['copyright']:
+            if byline:
+                byline = '{0} / {1}'.format(byline, entry['copyright'])
+            else:
+                byline = '|{0}'.format(entry['copyright'])
+
+        license_text = '{{%s%s}}' % (template, byline)
```

Comment:
I might actually restructure how this is done which would then also resolve the difference.